python/lsst/fgcmcal/fgcmBuildStarsTable.py

```diff
@@ -0,0 +1,347 @@
+from .utilities import computeApproxPixelAreaFields
+
+__all__ = ['FgcmBuildStarsTableConfig', 'FgcmBuildStarsTableTask']
+
+
+class FgcmBuildStarsTableConfig(FgcmBuildStarsConfigBase):
+    """Config for FgcmBuildStarsTableTask"""
+
+    referenceCCD = pexConfig.Field(
+        doc="Reference CCD for checking PSF and background",
+        dtype=int,
+        default=40,
+    )
+
+    def setDefaults(self):
+        self.instFluxField = 'ApFlux_12_0_instFlux'
+        self.localBackgroundFluxField = ''
+        self.apertureInnerInstFluxField = 'ApFlux_12_0_instFlux'
+        self.apertureOuterInstFluxField = 'ApFlux_17_0_instFlux'
+        self.psfCandidateName = 'Calib_psf_candidate'
+
+        sourceSelector = self.sourceSelector["science"]
+        sourceSelector.setDefaults()
+
+        fluxFlagName = self.instFluxField[0: -len('instFlux')] + 'flag'
+
+        sourceSelector.flags.bad = ['PixelFlags_edge',
+                                    'PixelFlags_interpolatedCenter',
+                                    'PixelFlags_saturatedCenter',
+                                    'PixelFlags_crCenter',
+                                    'PixelFlags_bad',
+                                    'PixelFlags_interpolated',
+                                    'PixelFlags_saturated',
+                                    'Centroid_flag',
+                                    fluxFlagName]
+
+        if self.doSubtractLocalBackground:
+            localBackgroundFlagName = self.localBackgroundFluxField[0: -len('instFlux')] + 'flag'
+            sourceSelector.flags.bad.append(localBackgroundFlagName)
+
+        sourceSelector.doFlags = True
+        sourceSelector.doUnresolved = True
+        sourceSelector.doSignalToNoise = True
+        sourceSelector.doIsolated = True
+
+        sourceSelector.isolated.parentName = 'parentSourceId'
+        sourceSelector.isolated.nChildName = 'Deblend_nChild'
+
+        sourceSelector.signalToNoise.fluxField = self.instFluxField
+        sourceSelector.signalToNoise.errField = self.instFluxField + 'Err'
+        sourceSelector.signalToNoise.minimum = 10.0
+        sourceSelector.signalToNoise.maximum = 1000.0
+
+        sourceSelector.unresolved.name = 'extendedness'
+        sourceSelector.unresolved.maximum = 0.5
+
+
+class FgcmBuildStarsTableTask(FgcmBuildStarsBaseTask):
+    """
+    Build stars for the FGCM global calibration, using sourceTable_visit catalogs.
+    """
+    ConfigClass = FgcmBuildStarsTableConfig
+    RunnerClass = FgcmBuildStarsRunner
+    _DefaultName = "fgcmBuildStarsTable"
+
+    @classmethod
+    def _makeArgumentParser(cls):
+        """Create an argument parser"""
+        parser = pipeBase.ArgumentParser(name=cls._DefaultName)
+        parser.add_id_argument("--id", "sourceTable_visit", help="Data ID, e.g. --id visit=6789")
+
+        return parser
+
+    def findAndGroupDataRefs(self, butler, dataRefs):
+        """
+        Find and group dataRefs (by visit).
```

Comment:
This is actually quite special purpose, the way I've written it (for good or ill).  The original src-catalog based `findAndGroupDataRefs` was written to try to avoid a lot of the pitfalls of Gen2, so that functionality is going to be obsoleted.  The new version is just to be a quick wrapper to be datatype-compatible so I could use my same codes.  And finally, with Gen3 I think that a lot of this jiggery pokery is going to go away (or at least be completely different).  As I start the pipeline-task-ification I'll make sure that the use cases I face will be general-purposed.